File: python/structural/facade/publisher/seo_optimIzer.py
def seo_optimizer(article: dict) -> dict:
    suggestions = {}
    
    if "keywords" not in article:
        suggestions["keywords"] = "Add relevant keywords"
    else:
        if len(article["keywords"]) < 3:
            suggestions["keywords"] = "Add at least 3 relevant keywords"

    if "meta_description" not in article:
        suggestions["meta_descriptions"] = "Add a meta description to the article"
    else:
        if len(article["meta_description"]) > 100 or len(article["meta_description"]) > 110:
            suggestions["meta_descriptions"] = "The meta description at least between 100 and 110 characters"
    
    if "internal_links" not in article or len(article["internal_links"]) == 0:
        suggestions["internal_links"] = "Add at least one internal link"

    return suggestions

File: typescript/src/structural/facade/publisher/SeoOptimizer.ts
import { Article, Suggestions } from "../interfaces/IArticle";

export function seoOptimizer(article: Article): Suggestions {
    const suggestions: Suggestions = {
        keywords: [],
        metaDescription: [],
        internalLinks: [],
    }

    if (!article.keywords){
        suggestions.keywords.push("Add relevant keywords")
    }
    else{
        if (article.keywords && article.keywords.length <= 3){
            suggestions.keywords.push("Add at least 3 relevant keywords")
        }
    }

    if (!article.metaDescription){
        suggestions.metaDescription.push("Add meta description to the article")
    }
    else{
        if (article.metaDescription && article.metaDescription.length <= 100 || article.metaDescription.length <= 110){
            suggestions.metaDescription.push("The meta description at least between 100 and 110 characters")
        }
    }

    if (!article.internalLinks || article.internalLinks.length == 0){
        suggestions.internalLinks.push("Add at least one internal link")
    }

    return suggestions
}